### PR TITLE
UXUI-618 - Adding btn-loading and loading as a class of properties

### PR DESCRIPTION
**Pull request by elliottcrush**
#### What does this PR do? (please provide any background)
This PR plugs the game left by Bootstrap when they deprecated the button loading features of `buttons.less`. This enables us to have a spinner 'on form submit' for example without compromising the width of the button. 

This change has markup that's dependent on it. So in order to have a 'loading' state on your button you need to have a span wrapping the button text together with an `<i>` tag holding the icon.

#### What tests does this PR have?
None

#### How can this be tested?
`npm link` into Pattern Library.

Or grab the `uxui-617` branch on the `static-site-generator-sites` and follow the steps to get a the `hx-customer-contact` template group running locally. After you've done that, navigate to `details-form.html` and click 'search'. You should see the spinner icon correctly aligned and the text should disappear.

#### Screenshots / Screencast
screen shot 2016-05-27 at 10 56 19
screen shot 2016-05-27 at 10 56 34

#### What gif best describes how you feel about this work?

---

- [x] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

#### Reviewers

**Review 1**
- [x] :+1:

**Review 2**
- [x] :+1:

**Review 3**
- [x] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

### CODE REVIEW

**Comment by nassimdonald** [comment]
🔎🔍

File: brands/holidayextras/src/less/_buttons.less [hunk 1]
```diff
@@ -21,6 +21,28 @@
     }
   }
 
+  &.btn-loading {
+    position: relative;
+    i {
+      visibility: hidden;
+      position: absolute;
+      width: 1em;
+      list-style: 1em;
```
Comment (nassimdonald) [inline brands/holidayextras/src/less/_buttons.less]: This should be `line-height: 1em`

**Comment by nassimdonald** [comment]
Wonderful 🐼

**Comment by jodiedoubleday** [comment]
:v: :shipit: 

NB: We should add this to UIToolkit at some point.

**Comment by jackdcrawford** [comment]
3️⃣  looks good 👍

**Pull request merged**